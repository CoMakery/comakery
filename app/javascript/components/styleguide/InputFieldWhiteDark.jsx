import React from 'react'
import classNames from 'classnames'
import InputField from './InputField'

class InputFieldWhiteDark extends React.Component {
  render() {
    const {
      className,
      ...other
    } = this.props

    const classnames = classNames(
      'input-field__white__dark',
      className
    )

    return (
      <React.Fragment>
        <InputField
          className={classnames}
          type="text"
          {...other}
        />
      </React.Fragment>
    )
  }
}

export default InputFieldWhiteDark
