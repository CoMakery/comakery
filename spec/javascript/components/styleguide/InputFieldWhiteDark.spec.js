import React from 'react'
import { shallow } from 'enzyme'
import InputFieldWhiteDark from 'components/styleguide/InputFieldWhiteDark'

describe('InputFieldWhiteDark', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<InputFieldWhiteDark />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.input-field__white__dark').props().type).toBe('text')
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<InputFieldWhiteDark className="__test" />)

    expect(wrapper.exists('.input-field__white__dark.__test')).toBe(true)
  })
})
