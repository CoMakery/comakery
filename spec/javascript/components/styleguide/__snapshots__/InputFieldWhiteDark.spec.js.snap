// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputFieldWhiteDark renders correctly without props 1`] = `
<Fragment>
  <InputField
    checkboxText=""
    checked={false}
    className="input-field__white__dark"
    disabled={false}
    errorText=""
    eventHandler={[Function]}
    imgPreviewUrl=""
    name="field"
    pattern=".*"
    placeholder="Please enter value"
    readOnly={false}
    required={false}
    selectEntries={Array []}
    symbolLimit={100}
    title="title"
    type="text"
    value=""
  />
</Fragment>
`;
